src/transformers/models/vit/feature_extraction_vit.py

```diff
@@ -0,0 +1,136 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Feature extractor class for ViT."""
+
+from typing import List, Union
+
+import numpy as np
+import torch
+from PIL import Image
+from torchvision import transforms as T
+
+from ...feature_extraction_utils import BatchFeature, FeatureExtractionMixin
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+class ViTFeatureExtractor(FeatureExtractionMixin):
+    r"""
+    Constructs a ViT feature extractor.
+
+    This feature extractor inherits from :class:`~transformers.FeatureExtractionMixin` which contains most of the main
+    methods. Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+        image_mean (:obj:`int`, defaults to [0.5, 0.5, 0.5]):
+            The sequence of means for each channel, to be used when normalizing images.
+        image_std (:obj:`int`, defaults to [0.5, 0.5, 0.5]):
+            The sequence of standard deviations for each channel, to be used when normalizing images.
+        do_normalize (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Whether or not to normalize the input with mean and standard deviation.
+        do_resize (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Whether to resize the input to a certain :obj:`size`.
+        size (:obj:`int`, `optional`, defaults to :obj:`224`):
+            Resize the input to the given size. Only has an effect if :obj:`do_resize` is set to :obj:`True`.
+    """
+
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self,
+        image_mean=[0.5, 0.5, 0.5],
+        image_std=[0.5, 0.5, 0.5],
+        do_normalize=True,
+        do_resize=True,
+        size=224,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.image_mean = image_mean
+        self.image_std = image_std
+        self.do_normalize = do_normalize
+        self.do_resize = do_resize
+        self.size = size
+
+    def __call__(
+        self,
+        images: Union[Image.Image, np.ndarray, torch.Tensor, List[Image.Image], List[np.ndarray], List[torch.Tensor]],
+        **kwargs
+    ) -> BatchFeature:
+        """
+        Main method to prepare for the model one or several image(s).
+
+        Args:
+            images (:obj:`PIL.Image.Image`, :obj:`np.ndarray`, :obj:`torch.Tensor`, :obj:`List[PIL.Image.Image]`, :obj:`List[np.ndarray]`, :obj:`List[torch.Tensor]`):
```

Comment:
It seems important to note that objects which are not `PIL.Image.Image` will be converted to PIL images, before being converted back to torch tensors.

Reply:
Also, this feature extractor returns torch tensors. I think one of the expected behaviors of our pre-processors (tokenizers or feature extractors) are to be framework-agnostic, even if no other framework is supported.

The idea is that even if users don't use our implementation of ViT, they can still benefit from the feature extractor.

At the very least it should be explicitely mentioned in the documentation that it will return torch tensors, but I'd much rather it be framework agnostic and have a `return_tensors` flag!

Reply:
Yes, currently it's not framework-agnostic, as discussed on Slack. We are adding only ViT in PyTorch for now, and we are using `torchvision` for the image transformations. In the future, one will be able to use the `return_tensors` flag. 

I'll add a `returns` docstring.

Reply:
Maybe add a clear warning to tell the users the processor will convert numpy arrays and torch tensors to PIL images so the most efficient is to pass PIL images?